tracing-subscriber/tests/duplicate_spans.rs

```diff
@@ -0,0 +1,37 @@
+mod support;
+use self::support::*;
+use tracing::{self, subscriber::with_default, Level, Span};
+use tracing_subscriber::{filter::EnvFilter, prelude::*, FmtSubscriber};
+
+#[test]
+fn duplicate_spans() {
+    let subscriber = FmtSubscriber::builder()
+        .with_env_filter(EnvFilter::new("[root]=debug"))
+        .finish();
+
+    with_default(subscriber, || {
+        let root = tracing::debug_span!("root");
+        let _enter = root.enter();
+        // root:
+        assert_eq!(root, Span::current());
```

Comment:
Take it or leave it: it _might_ be helpful in the future to add messages to these assertions saying what spans have been entered. Not a big deal though.

Reply:
ok